### PR TITLE
Split ui package into utils and azureutils packages

### PR DESCRIPTION
**Pull request by alexweininger**
Working towards #1051 

### Work timeline

* Create @microsoft/vscdoe-azext-azureutils package **done in this PR**
* Rename @vscode-azureextensionui to @microsoft/vscode-azext-utils **done in this PR**
* Publish both packages to npm, and begin migrating extensions
* Rename utils (ui package) exports  to more generic names. ex: `AzureNameStep` → `AzExtNameStep`
* Further refactoring if needed...

### Changes

`ui/src/index.ts` shows a nice summary of what's been relocated (see `ui/index.d.ts` for more detailed summary).

The following has moved from ui to new azure utils package:
* createAzureClient
* openInPortal
* AzureAccountTreeItemBase
* SubscriptionTreeItemBase
* LocationListStep
* ResourceGroupCreateStep
* ResourceGroupListStep
* ResourceGroupNameStep
* StorageAccountCreateStep
* StorageAccountListStep
* StorageAccountNameStep
* VerifyProvidersStep
* uiUtils (listAllIterator)

 The following have been exported from ui and imported by azure utils package:
* openUrl
* addExtensionValueToMask
* callWithMaskHandling
* maskValue
* nonNull utils

And I added `registerAzureUtilsExtensionVariables` which is a carbon copy of `registerAppServiceExtensionVariables`.

### Migrating

I've migrated and tested locally Docker and VMs to use both of the new packages. See what those changes look like:
* https://github.com/microsoft/vscode-docker/compare/alex/azureUtils?expand=1
* https://github.com/microsoft/vscode-azurevirtualmachines/compare/main...alex/t2-split?expand=1 **updated**

### Notes

I plan for the new `@microsoft/vscode-azext-utils` package to live in `utils`. However, I've left the changes in the `ui` folder so you can see diff on GitHub. Once approved I'll revert the changes made in the `ui` folder. That way we're left with an unchanged version of the `vscode-azureextensionui` package code in case we need to make fixes/changes in there later.

### CODE REVIEW

File: azure/.npmignore [hunk 1]
```diff
@@ -0,0 +1,13 @@
+node_modules/
```
Comment (bwateratmsft) [inline azure/.npmignore:1]: I'm guessing some of the resources (like the .PNG's) should also be in this file. Probably _not_ the SVG for the icon though.

File: utils/.npmignore [hunk 2]
```diff
@@ -0,0 +1,13 @@
+node_modules/
```
Comment (bwateratmsft) [inline utils/.npmignore:1]: Same here WRT resources e.g. the PNGs

File: utils/src/utils/nonNull.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+ *  Licensed under the MIT License. See License.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { isNullOrUndefined } from 'util';
+
+/**
+ * Retrieves a property by name from an object and checks that it's not null and not undefined.  It is strongly typed
+ * for the property and will give a compile error if the given name is not a property of the source.
+ */
+export function nonNullProp<TSource, TKey extends keyof TSource>(source: TSource, name: TKey): NonNullable<TSource[TKey]> {
```
Comment (bwateratmsft) [inline utils/src/utils/nonNull.ts:12]: These functions are repeated in a lot of our packages/extensions; can we just export them from here?

**Review by bwateratmsft — APPROVED** [review]

**Comment by alexweininger** [comment]
I plan on waiting for #1050 to be merged so I can pick up those changes.

File: utils/.npmignore (hunk 2 above)
Comment (alexweininger) [reply utils/.npmignore:1]: Do you know if the PNGs will still show up on the readme on npm?

Comment (bwateratmsft) [reply utils/.npmignore:1]: I guess I'm not certain. We could test without them in the package and just update it if they don't show up.

**Comment by bwateratmsft** [comment]
It's not related but after this PR is finished is probably a good time to fix #978.

File: .azure-pipelines/package-steps.yml [hunk 4]
```diff
@@ -45,7 +45,7 @@ steps:
 - task: CopyFiles@2
   displayName: '${{ parameters.package }}: copy package'
   inputs:
-    Contents: '${{ parameters.package }}/vscode-*.tgz'
+    Contents: '${{ parameters.package }}/@(microsoft|vscode)-*.tgz'
```
[the remark this answers is not in the document]
Comment (alexweininger) [reply .azure-pipelines/package-steps.yml]: It's actually specific file matching syntax required for the CopyFiles@2 task.

https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/file-matching-patterns?view=azure-devops#asterisk_examples

File: utils/package.json [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+        "dayjs": "^1.9.3",
+        "escape-string-regexp": "^2.0.0",
+        "html-to-text": "^5.1.1",
+        "open": "^8.0.4",
+        "semver": "^5.7.1",
+        "vscode-nls": "^4.1.1",
+        "vscode-tas-client": "^0.1.22"
+    },
+    "devDependencies": {
+        "@azure/core-client": "^1.5.0",
```
[the remark this answers is not in the document]
Comment (alexweininger) [reply utils/package.json]: Yep, I moved those and the uiUtils that needed these types to the azure package. So now the only azure related (dev) dependency in the utils package is `@azure/ms-rest-azure-env` which we need for `ISubscriptionContext`.

**Pull request merged**